libsolidity/Compiler.cpp

```diff
@@ -247,6 +249,16 @@ void Compiler::registerStateVariables(ContractDefinition const& _contract)
 			m_context.addStateVariable(*variable);
 }
 
+void Compiler::initializeStateVariables(ContractDefinition const& _contract)
+{
+	for(ASTPointer<VariableDeclaration> const& variable: _contract.getStateVariables())
+		if (variable->getValue())
+		{
+			compileExpression(*(variable->getValue()), (variable->getValue())->getType());
```

Comment:
If you invoke the expression compiler in the next line anyway, why not move this line into `appendStateVariableInitialization`?